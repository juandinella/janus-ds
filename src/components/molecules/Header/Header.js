import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './Header.styles'

import CloseButton from '../../atoms/CloseButton'
import Text from '../../atoms/Text'

/**
 * Header
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledHeader = styled.header.withConfig({ shouldForwardProp })`
  ${styles}
`

const Header = ({ id, text, onClick = () => null }) => {
  return (
    <StyledHeader id={id} data-testid={id}>
      <Text weight="medium" size="sm" color="500">
        {text}
      </Text>
      <CloseButton onClick={onClick} />
    </StyledHeader>
  )
}

Header.displayName = 'Header'

Header.propTypes = {
  id: PropTypes.string,
  text: PropTypes.node,
  onClick: PropTypes.func,
}

export default Header
