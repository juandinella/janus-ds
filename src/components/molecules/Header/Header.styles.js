import { css } from 'styled-components'

export default css`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: var(--spacing-xs);
  border-radius: var(--border-radius-lg) var(--border-radius-lg) 0 0;
`
