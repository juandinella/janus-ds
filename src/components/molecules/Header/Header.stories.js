import Header from './Header'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(Header)

export default {
  title: 'Molecules/Header',
  component: Header,
  args: {
    text: 'Chat with Janus',
  },
  argTypes: {},
}

export const Default = Template.bind({})
