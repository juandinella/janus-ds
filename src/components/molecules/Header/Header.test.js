import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import Header from './Header'

initStoryshots({
  storyKindRegex: /^Molecules\/Header$/,
})

const componentProps = {
  id: 'unique_id',
  text: 'Go!',
}

afterEach(cleanup)

describe('<Header />', () => {
  test('should render component correctly', () => {
    render(<Header {...componentProps} />)
    const header = screen.getByTestId(componentProps.id)
    expect(header).toBeInTheDocument()
  })

  test('should render content correctly', () => {
    render(<Header {...componentProps} />)
    const p = screen.getByText(componentProps.text)
    expect(p).toBeInTheDocument()
  })
})
