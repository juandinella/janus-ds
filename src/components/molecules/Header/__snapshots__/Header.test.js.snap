// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/Header Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <header
    className="sc-i6tidb-0 llEsGp"
  >
    <md
      class="sc-81qx22-0 fWZQyn"
      color="500"
      size="sm"
    >
      Chat with Janus
    </md>
    <button
      className="sc-x4jc1v-0 cOgVEl"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="14"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M13.707.293a1 1 0 010 1.414l-12 12a1 1 0 01-1.414-1.414l12-12a1 1 0 011.414 0z"
          fill="var(--color-neutral-500)"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M.293.293a1 1 0 011.414 0l12 12a1 1 0 01-1.414 1.414l-12-12a1 1 0 010-1.414z"
          fill="var(--color-neutral-500)"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="sc-90c4yq-0 bmmsAZ"
      >
        Close Chat
      </span>
    </button>
  </header>
</div>
`;
